packages/driver/src/cy/commands/actions/attach-file-tech-brief/README.md

```diff
@@ -42,21 +41,21 @@ This triggers `input` and `change` events on the given input element, signaling
 It also supports arrays of files, representing a user selecting multiple files at once. If the input element does not have `multiple` property, an error will be thrown.
 
 ```js
-  cy.get('input[type=file]').attachFile(['fixture:users.json', '@myFile'])
+  cy.get('input[type=file]').attachFile(['users.json', '@myFile'])
 ```
 
 For more precise control, a user can pass in an object (or array of objects) giving details about the files:
 ```js
   cy.get('input[type=file]').attachFile({
     contents: '@myBuffer',
-    filePath: '/imaginary/path/to/file.txt',
+    fileName: '/imaginary/path/to/file.txt',
     lastModified: Date.now(),
     mimeType: 'application/octet-stream',
   })
 ```
 
-- `contents` can be any of the above formats - `fixture:file.json`, an `@alias`, or a `Buffer()`. Required.
-- `filePath` is a string. If undefined, it defaults to the actual path (if we have one from `cy.readFile()` or `cy.fixture()`), or an empty string if none can be inferred.
+- `contents` can be any of the above formats - `file.json`, an `@alias`, a `Buffer()`, or an object to JSON.stringify(). Required.
```

Comment:
I see why you want to re-use `contents` for multiple different meaning, even when it's in an object, I think. Is it so that `cy.attachFile(contents)` and `cy.attachFile({ contents })` work exactly the same?

Option B would've been to let the user use separate, mutually-exclusive keys in this object... `fileName`, `alias`, `contents` (which could still work with strings or Buffers or objects). This would be a little easier to TSify and debug, I think.

Can you comment on why the API was designed this way instead of option B?

Reply:
Exactly so. Consider the following:
```
cy.attachFile('fixtures/foo.json')

cy.readFile('fixtures/image.png', { encoding: null }).then((image) => {
  // Process image somehow
  cy.attachFile(image)
})
```

Based on how cypress-upload-file is used, I suspect these two cases will be by far the most common, and therefore wanted to make them as simple as possible, where you can pass in exactly what you want as a single argument and it "just works". That means that the same argument has to be interpreted in a couple of different ways - and since we're already parsing out what to do when *not* passed in an object, it seemed natural to use the same logic on the `contents` property when we do have an object.

Basically - option B is indeed nicer for the case when an object is passed in. But since we already have the logic for differentiating the type of the argument when it's not an object, I chose consistency.